This is a proposal for SUID Enumeration Using Find

mlakri opened the pull request:
<!--
Thanks for your contribution. Please make sure to fill the contents of this template with the necessary information to ease and speed up the review process.

!!! PLEASE DO NOT DELETE ANY SECTION, COMMENT OR THE CONTENT OF THE TEMPLATE. !!!
-->

The goal is to detect the execution of the find command in auditd logs (e.g., $find /usr/bin -type f -perm -4001) when searching for binaries or scripts that can be executed with elevated privileges, or when used for reconnaissance to discover details about file permissions or access conditions to sensitive data.

<!--
**Please note that this section is required and must be filled**
A short summary of your pull request.
-->

### Changelog

<!--
** Don't remove this comment **
You need to add one line for every changed file of the PR and prefix one of the following tags:
new:	<title>
update:	<title> - <optional comment>
fix:	<title> - <optional comment>
remove:	<title> - <optional comment>
chore: for non-detection related changes (e.g. dates/titles) and changes on workflow

e.g.
new: Brute-Force Attacks on Azure Admin Account
update: Suspicious Microsoft Office Child Process - add MSPUB.EXE
fix: Malware User Agent - remove legitimate Firefox UA
chore: workflow - update checkout version
remove: Suspicious Office Execution - deprecated in favour of 8f922766-a1d3-4b57-9966-b27de37fddd2
-->

type=EXECVE msg=audit(1731503214.337:71048): argc=6 a0="find" a1="/usr/bin" a2="-type" a3="f" a4="-perm" a5="-4001"

<!--
Fill this in case of false positive fixes
-->

### Fixed Issues

<!--
Link the fixed issues here, in case your commit fixes issues with rules or code
-->

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)

mlakri committed:
This is a proposal for a PrivEsc detection rule

The idea is to detect in the auditd logs the execution of the find command (for example, $find -perm -u=ws ...) looking for binaries or scripts that can be executed with elevated privileges.

nasbench commented:
Hey @mlakri and thanks for your contribution.

Can you please provide log data for your example. And also looking for specific permission isn't a direct sign of privilege escalation. It's just permission discovery, the outcome could be exfiltration, recon or privesc for privesc. So the metadata fro the rule (description and title) should reflect that.

Please take this into consideration and make the necessary changes.

mlakri committed:
Update lnx_auditd_enum_binaries_suid.yml

mlakri commented:
Hello @nasbench 
I removed everything related to "PrivEsc," leaving its meaning to refer only to enumeration. 
Additionally, I added an example of an EXECVE log.

mlakri committed:
Create lnx_auditd_deletion_auditd_rules.yml

mlakri committed:
Merge pull request #1 from mlakri/MyRules-1

This rule detects the deletion of existing Auditd rules

nasbench committed:
Delete rules/linux/auditd/lnx_auditd_deletion_auditd_rules.yml

swachchhanda000 reviewed (commented):
Closing bcz of no activity and we do have some coverage in regards of susp suid enumeration.
https://grep.app/search?f.repo=SigmaHQ%2Fsigma&f.repo.pattern=sigma&q=-perm
